### ISSUE TITLE
PermitIf + Parameterized triggers

### ISSUE THREAD

**Issue by tastyeggs** [prasannavl/LiquidState#26]
PermitIf must expose the trigger parameter in its predicate for them to be useful. Since config lacks context (as I want to make that a singleton throughout my app), I'm trying to pass in context as a trigger parameter, and would like to add guard conditions that involve this context.

Would suggest an overload such as this:

`PermitIf<TArgument>(Func<TArgument,Task<bool>> predicate, ParameterizedTrigger<TTrigger, TArgument> trigger, TState resultingState, Action<Transition<TState, TTrigger>, TArgument> onTriggerAction)`

**Comment by alexmarshall132** [comment]
I agree. Not being able to pass the argument to the predicate makes no sense and makes using this library much harder and more complicated.

**Comment by prasannavl** [comment]
I'll look into it tonight.

**Comment by prasannavl** [comment]
@tastyeggs  - This is something that I'd like to work on. However I'm not sure I'll get the time to this in the next few days. So, might take a while. 

Meanwhile, I understand there is a need. So, I'll happy to take in any PRs into the dev branch.

**Comment by alexmarshall132** [comment]
Thanks Prasanna. That would be great.

On Sat, Feb 27, 2016 at 8:08 PM, Prasanna V. Loganathar <
notifications@github.com> wrote:

> @tastyeggs <https://github.com/tastyeggs> - This is something that I'd
> like to work on. However I'm not sure I'll get the time to this in the next
> few days. So, might take a while.
>
> Meanwhile, I understand there is a need. So, I'll happy to take in any PRs
> into the dev branch.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/prasannavl/LiquidState/issues/26#issuecomment-189772546>
> .
>



-- 
Alex
